src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitWorkerService.java

```diff
@@ -143,19 +143,23 @@ private List<Fault> getAllBusfaultFromNetwork(Network network, ShortCircuitRunCo
         return faults;
     }
 
-    private List<Fault> getBusFaultFromBusId(Network network, ShortCircuitRunContext context) {
+    private List<Fault> getBusFaultFromBusId(Network network, ShortCircuitRunContext context, UUID resultUuid) {
         String busId = context.getBusId();
         Identifiable<?> identifiable = network.getIdentifiable(busId);
         Map<String, ShortCircuitLimits> shortCircuitLimits = new HashMap<>();
 
-        if (identifiable instanceof BusbarSection) {
-            String busIdFromBusView = ((BusbarSection) identifiable).getTerminal().getBusView().getBus().getId();
+        if (identifiable instanceof BusbarSection busbarSection) {
+            Bus bus = busbarSection.getTerminal().getBusView().getBus();
+            if (bus == null) {
+                notificationService.publishFail(resultUuid, context.getReceiver(), "Selected bus is out of voltage", context.getUserId(), busId);
```

Comment:
You can replace those two rows by something like : 
`throw new PowsyblException("Selected bus" + busId + " is out of voltage");`

So you're homogeneous with row 173, you don't need to give resultUuid as parameter and you can remove the change at row 191